mlrun/db/httpdb.py

```diff
@@ -2586,9 +2586,24 @@ def _validate_version_compatibility(server_version, client_version) -> bool:
                 parsed_client_version=parsed_client_version,
             )
             return False
-        if parsed_server_version.minor != parsed_client_version.minor:
+        if parsed_server_version.minor > parsed_client_version.minor + 2:
             logger.info(
-                "Server and client versions are not the same",
+                "Backwards compatibility might not apply between the server and client version",
+                parsed_server_version=parsed_server_version,
+                parsed_client_version=parsed_client_version,
+            )
+            return False
+        if parsed_client_version.minor > parsed_server_version.minor:
+            logger.warning(
+                "Client version with higher version than server version isn't supported,"
+                " align your client to the server version",
+                parsed_server_version=parsed_server_version,
+                parsed_client_version=parsed_client_version,
+            )
+            return False
+        if parsed_server_version.minor <= parsed_client_version.minor + 2:
```

Comment:
```suggestion
        if parsed_server_version.minor != parsed_client_version.minor:
```

This check is not accurate - if `server.minor` is the same as `client.minor` this will still hold, and you'll get a warning that they're not the same even if they are...